community/server/src/test/java/org/neo4j/server/rest/transactional/ExecutionResultSerializerTest.java

```diff
@@ -82,14 +83,30 @@
 import static org.neo4j.test.mocking.GraphMock.relationship;
 import static org.neo4j.test.mocking.Properties.properties;
 
-public class ExecutionResultSerializerTest
+public class ExecutionResultSerializerTest extends TxStateCheckerTestSupport
```

Comment:
Do we have any tests covering the nested cases? 

    MATCH (n) DELETE (n) RETURN [n, {someKey: n}]

Also, I'm still unclear on the use case here - what is the remote client supposed to do with the 'isDeleted' flag? Similar for the `id` that gets returned, the use case seems unclear. Since 99% applications in the wild use labels/properties for identity, what are they supposed to do with that lone `id`?